### PR TITLE
add feature importances for RandomForest

### PR DESCRIPTION
**Pull request by LiuYuHui**

### CODE REVIEW

**Review by vigsterkr — CHANGES_REQUESTED** [review]
great! could you please add a `watch_method` so that this is exposed via the parameter framework.

File: src/shogun/machine/RandomForest.h [hunk 1]
```diff
@@ -137,6 +137,12 @@ class RandomForest : public BaggingMachine
 	 * @return number of randomly chosen features during each node split
 	 */
 	int32_t get_num_random_features() const;
+	/** get feature importances of previous trained, use Mean Decrease
+	 * Impurity(MDI)
+	 *
+	 * @return arrays of feature importance
+	 */
+	SGVector<float64_t> get_feature_importances();
```
Comment (gf712) [inline src/shogun/machine/RandomForest.h]: If you don't cache the feature importances you could make this `const`, no?

Comment (LiuYuHui) [reply src/shogun/machine/RandomForest.h]: ok,i will change it :)

File: src/shogun/machine/RandomForest.cpp [hunk 2]
```diff
@@ -182,10 +183,32 @@ bool RandomForest::train_machine(std::shared_ptr<Features> data)
 	return BaggingMachine::train_machine();
 }
 
+SGVector<float64_t> RandomForest::get_feature_importances() const
```
Comment (karlnapf) [inline src/shogun/machine/RandomForest.cpp:186]: minor: is the name "get" appropriate, rather than "compute"?

**Pull request merged**